### ISSUE TITLE
Feature Request: API or socket to send SMS

### ISSUE THREAD

**Issue by do6uk** [g4klx/MMDVMHost#391]
Dear Jonathan,

first of all sorry for my bad english and many thanks for your work. We are using MMDVMHost on a repeater and want to do following:

**primary:** send SMS on RF-side to user and/or talkgroup on relevant events (e.g. lost connection to internet, critical system-state) generated/initiated by local scripts

_secondary:_ send SMS to DMR-Network-side like above.

how it shold work:
- script is calling an API or socket in MMDVMHost 
- giving **source-id**, **route** (rf, net, both), **type** (private/group). **target-id** and **text**
- MMDVMHost generates data and injects in Network-Stream if **route** is _net_ or _both_ and sending via rf if **route** is _rf_ or _both_

Do you think this could be possible?

**Comment by g2rHP7prvJqO** [comment]
I second this at 500%. It would be great to have simple SMS send function via modem (and network for Brandmeister only) to transmit telemetry information.
This might be potentially achieved with a hack by "forging" a UDP message from Brandmeister network to force the modem to transmit but a real API/socket would be 10x better.

**Comment by narspt** [comment]
+1, it would be a nice feature indeed, as well as it would be also nice if mmdvmhost was able to decode SMS, both would allow to create interactive remote things like send specific SMS to get specific information back on another SMS.

**Comment by Yentel** [comment]
+1 Target-id could be a dynamic list of all talkgroups that the repeater was subscribed to on the moment it lost connection, so that everyone that is QRV on the repeater receives the (group) text, and not only the sysop :-)

**Comment by on7lds** [comment]
+1 code/decode SMS would get us al lot of possibilities for interfacing from and to the digital networks. This could be the area of systemcontrol (see Yentel's example, but also remote start, restart, GPIOinterfacing for controlling repeater peripheral device, ...)  and surely also for DIY projects [I already have some in mind :-) ]

**Comment by Yentel** [comment]
Watch out with remote control, dmr id's can be changed easily so one can
control a repeater. You'd need to send an auth code specified in the config
to avoid this (or 2fa codes from google, but that requires internet on the
repeater)

**Comment by on7lds** [comment]
For critical remote control I was thinking about using an OTP (if it's a 'production' repeater).
For home-hotspots it could be no password or a very simple method.
There are simple repeater control implementations (like https://github.com/g4klx/MMDVMHost/blob/master/linux/tg_shutdown.sh) which have also no security (apart from checking the sender's ID) . It all boils down to how much of a risk of abuse there is.
In short, I see no unbeatable problems in that part.
Implementing the messages on the other hand...  (it would have to provide solutions for ETSI, Hytera and Chinese radios)

**Comment by narrowfail** [comment]
+1

**Comment by g4klx** [comment]
There is no standard for SMS in DMR. So how would you know what to send over the air? It depends on the radio. This is way too complex for the MMDVM end, better for someone like BM or DMR+ to implement it.
